Lib/fontTools/otlLib/builder.py

```diff
@@ -2573,7 +2578,8 @@ def buildLigGlyph(coords, points):
     # ([500], [4]) --> otTables.LigGlyph; None for empty coords/points
     carets = []
     if coords:
-        carets.extend([buildCaretValueForCoord(c) for c in sorted(coords)])
+        coords = sorted(coords, key=lambda c: isinstance(c, tuple) and c[0] or c)
```

Comment:
if c[0] == 0 the `and c[0]` test would evaluate to False and the `or c` would kick in, so the tuple as a whole instead of its first item would involutarily be passed on. I think you meant something like this:

```suggestion
        coords = sorted(coords, key=lambda c: c[0] if isinstance(c, tuple) else c)
```